File: C++/프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(vector<string> survey, vector<int> choices) {
    int score[26] = {};   // 'A'~'Z', 반드시 0으로 초기화

    for (size_t i = 0; i < survey.size(); i++) {
        int c = choices[i];
        if (c < 4)      score[survey[i][0] - 'A'] += 4 - c;
        else if (c > 4) score[survey[i][1] - 'A'] += c - 4;
    }

    string answer;
    answer.reserve(4);
    for (const char* pair : {"RT", "CF", "JM", "AN"}) {
        char a = pair[0], b = pair[1];
        answer += (score[a - 'A'] >= score[b - 'A']) ? a : b;
    }
    return answer;
}

File: Java/프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.java
class Solution {
    public String solution(String[] survey, int[] choices) {
        int[] score = new int[26];   // 'A'~'Z' 인덱스

        for (int i = 0; i < survey.length; i++) {
            int c = choices[i];
            if (c < 4)      score[survey[i].charAt(0) - 'A'] += 4 - c;
            else if (c > 4) score[survey[i].charAt(1) - 'A'] += c - 4;
        }

        StringBuilder sb = new StringBuilder();
        String[] pairs = {"RT", "CF", "JM", "AN"};
        for (String p : pairs) {
            char a = p.charAt(0), b = p.charAt(1);
            sb.append(score[a - 'A'] >= score[b - 'A'] ? a : b);
        }
        return sb.toString();
    }
}